src/main/ruby/core/truffle/interop.rb

```diff
@@ -24,17 +24,73 @@ def self.import_method(name)
     def self.export_method(name)
       export(name.to_s, Object.method(name.to_sym))
     end
+    
+    def self.object_has_keys(object)
+      case object
+      when NilClass, TrueClass, FalseClass, Fixnum, Bignum, Float, Symbol
+        false
+      else
+        true
+      end
+    end
 
-    def self.object_keys(object)
+    def self.object_keys(object, internal)
       if object.is_a?(Hash)
-        object.keys.map(&:to_s)
+        keys = object.keys.map(&:to_s)
       else
-        object.instance_variables.map { |ivar|
+        keys = object.instance_variables.map { |ivar|
           ivar = ivar.to_s
           ivar = ivar[1..-1] if ivar.start_with?('@')
           ivar
         }
       end
+      if internal
+        object.instance_variables.each do |ivar|
+          ivar = ivar.to_s
+          if ivar.start_with?('@')
+            keys << ivar
+          end
+        end
+      end
+      keys
+    end
+    
+    def self.key_info(object, name)
+      key_info_flags_from_bits(key_info_bits(object, name.to_sym))
+    end
+    
+    def self.object_key_info(object, name)
+      existing, readable, writable, invocable, internal = false, false, false, false, false
+      if name.to_s[0] == '@'
+        writable = true unless object.frozen?
+        internal = true
+        if object.instance_variable_defined?(name)
+          existing = true
+          readable = true
+        end
+      elsif object.is_a?(Hash)
+        if object.key?(name)
+          existing = true
+          readable = true
+          writable = true unless object.frozen?
+        end
+      else
+        name = name.to_sym
+        readable = object.respond_to?(name)
+        writable = object.respond_to?((:"#{name}=").to_sym) && !object.frozen?
```

Comment:
No need for `to_sym` here. `:"#{name}="` is a Symbol literal.

Reply:
Done.